instrumentation-tests/src/androidTest/java/com/mapbox/navigation/instrumentation_tests/core/WaypointsTest.kt

```diff
@@ -0,0 +1,256 @@
+        "J Street" to ApproximateCoordinates(38.576795, -121.480256, tolerance),
+        "C Street" to ApproximateCoordinates(38.582195, -121.468458, tolerance),
+    )
+    private val expectedSecondNonEvWaypointsNamesAndLocations = listOf(
+        "9th Street" to ApproximateCoordinates(38.577764, -121.496066, tolerance),
+        "J Street" to ApproximateCoordinates(38.576795, -121.480256, tolerance),
+        "C Street changed for tests" to ApproximateCoordinates(38.582195, -121.468458, tolerance),
+    )
+
+    override fun setupMockLocation(): Location = mockLocationUpdatesRule.generateLocationUpdate {
+        latitude = nonEvCoordinates[0].latitude()
+        longitude = nonEvCoordinates[0].longitude()
+        bearing = 190f
+    }
+
+    @Before
+    fun setup() {
+        runOnMainSync {
+            mapboxNavigation = MapboxNavigationProvider.create(
+                NavigationOptions.Builder(activity)
+                    .accessToken(getMapboxAccessTokenFromResources(activity))
+                    .routingTilesOptions(
+                        RoutingTilesOptions.Builder()
+                            .tilesBaseUri(URI(mockWebServerRule.baseUrl))
+                            .build()
+                    )
+                    .navigatorPredictionMillis(0L)
+                    .build()
+            )
+            mockWebServerRule.requestHandlers.clear()
+        }
+    }
+
+    @Test
+    fun ev_route_with_waypoints_in_response_root_by_default() = sdkTest {
+        addResponseHandler(R.raw.ev_route_response_with_waypoints_in_root, evCoordinates)
+        val routes = requestRoutes(evCoordinates, electric = true, waypointsPerRoute = null)
+
+        checkWaypointsInRoot(expectedEvWaypointsNamesAndLocations, routes[0])
+    }
+
+    @Test
+    fun ev_route_with_waypoints_in_response_root() = sdkTest {
+        addResponseHandler(R.raw.ev_route_response_with_waypoints_in_root, evCoordinates)
+        val routes = requestRoutes(evCoordinates, electric = true, waypointsPerRoute = false)
+
+        checkWaypointsInRoot(expectedEvWaypointsNamesAndLocations, routes[0])
+    }
+
+    @Test
+    fun ev_route_with_waypoints_per_route() = sdkTest {
+        addResponseHandler(R.raw.ev_route_response_with_waypoints_per_route, evCoordinates)
+        val routes = requestRoutes(evCoordinates, electric = true, waypointsPerRoute = true)
+
+        checkWaypointsPerRoute(expectedEvWaypointsNamesAndLocations, routes[0])
+    }
+
+    @Test
+    fun non_ev_route_with_waypoints_in_response_root_by_default() = sdkTest {
+        addResponseHandler(R.raw.route_response_with_waypoints_in_root, nonEvCoordinates)
+        val routes = requestRoutes(nonEvCoordinates, electric = false, waypointsPerRoute = null)
+
+        checkWaypointsInRoot(expectedFirstNonEvWaypointsNamesAndLocations, routes[0])
+        checkWaypointsInRoot(expectedFirstNonEvWaypointsNamesAndLocations, routes[1])
+    }
```

Comment:
There're two routes but only one configured route refresh response. Is it expected that both routes will be updated with the same data?